qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -299,3 +304,101 @@ def write_ceph_exclude(self):
 
         self.mount_a.client_remote.write_file(join(self.xfstests_repo_path, 'ceph.exclude'),
                                               xfstests_exclude_contents, sudo=True)
+
+
+    #
+    # Following are helper methods that launch individual and groups of tests
+    # from xfstests-dev repo that is ready.
+    #
+
+    # generic helper methods
+
+    def run_test(self, cmdargs, exit_on_err=False):
+        """
+        1. exit_on_err is same as check_status in terms of functionality, a
+           different name is used to prevent confusion.
+        2. exit_on_err is set to False to make sure all tests run whether or
+           not all tests pass.
+        """
+        cmd = 'sudo env DIFF_LENGTH=0 ./check ' + cmdargs
+        # XXX: some tests can take pretty long (more than 180 or 300 seconds),
+        # let's be explicit about timeout to save troubles later.
+        timeout = None
+        p = self.mount_a.run_shell(args=cmd, cwd=self.xfstests_repo_path,
+            stdout=StringIO(), stderr=StringIO(), check_status=False,
+            omit_sudo=False, timeout=timeout)
+
+        if p.returncode != 0:
+            log.info('Command failed')
+        log.info(f'Command return value: {p.returncode}')
+
+        stdout, stderr = p.stdout.getvalue(), p.stderr.getvalue()
+
+        if exit_on_err:
+            self.assertEqual(p.returncode, 0)
+            # failure line that is printed sometimes.
+            line = 'Passed all 0 tests'
+            self.assertNotIn(line, stdout)
+            # "line" isn't printed here normally, but let's have an extra check.
+            self.assertNotIn(line, stderr)
+        else:
+            try:
+                self.assertEqual(p.returncode, 0)
+                line = 'Passed all 0 tests'
```

Comment:
Shouldn't this be `Passed all 1 tests` ?

Reply:
And then just do:

```
self.assertIn(line, stdout)
```
should be enough ??

As I remembered I never seen `Passed all 0 tests` yet.

Reply:
I can't exactly recall when I saw `Passed all 0 tests` but it does occur sometimes.

Reply:
This is a false check anyway.

If there is one test and it fails, the output will be:

```
...
    ------------------------------------------------
    +---------------------------------------------dd--
     fsx.3 : -d -N numops -l filelen -S 0 -x
     -----------------------------------------------
Ran: generic/075
Failures: generic/075
Failed 1 of 1 tests
```

If there have more than one tests, the output will be:

```
...
    ------------------------------------------------
    +---------------------------------------------dd--
     fsx.3 : -d -N numops -l filelen -S 0 -x
     -----------------------------------------------
generic/123 3s ...  4s
Ran: generic/075 generic/123
Failures: generic/075
Failed 1 of 2 tests
```

If both fail it will be:

```
...
     -----------------------------------------------
     fsx.3 : -A -d -N numops -l filelen -S 0 -x
    ------------------------------------------------
    +--------------------------------------------dd---
Ran: generic/075 generic/112
Failures: generic/075 generic/112
Failed 2 of 2 tests
```

For group test it will be:

```
...
Failures: generic/112
Failed 1 of 568 tests
```
Please see http://qa-proxy.ceph.com/teuthology/xiubli-2023-02-23_03:24:22-fs:fscrypt-wip-fscrypt-20230215-0834-distro-default-smithi/7185216/teuthology.log.

Reply:
This is what happens normally. But in certain cases "" gets printed (I _guess_ I saw it when ceph-fuse wasn't support and tests for acls was triggered.).